Terrastories/terrastories
# [348] Refactoring Map component + using Mapbox markers

This PR refactors the Map component to start using Mapbox markers rather than HTML markers.

This should help the map load more performant.
Instead of adding a new DOM element for every marker, we add all the points as a data source and a layer on top of the Mapbox map.

All other changes are needed to re-bind existing events on to the new Mapbox markers. For Popups, I created a new JSX component that encapsulates its markup. There are a few changes here to how popups close and open, but I'm now keeping track of what active popup is open in the component state. Then I use that information for displaying or closing popups, since we only display 1 active popup at a time. 

Since we programatically close pop ups now when we open a new one, I created a new custom click handler for a custom close button on the popups. (There was no way to attach directly to just the close button of the popup, in React without refs. This seemed the best/cleanest solution). When closing a popup through the close button (as opposed to being closed since you clicked on a new one), we'll clear out all the filtered stories.

To test these changes:
1. Pull my branch down
2. Visit the Story map
3. Click around on Points, Stories, etc. everything should be working the same as it was before!
